Source/WebCore/dom/MouseEvent.idl

```diff
@@ -44,13 +44,11 @@
     readonly attribute double movementY;
     [ImplementedAs=force] readonly attribute double webkitForce;
 
-    // FIXME: relatedTarget should be of type EventTarget? but we need to use any to support a quirk for iAd Producer (rdar://problem/30640101).
-    [CallWith=CurrentGlobalObject&CurrentScriptExecutionContext, ImplementedAs=initMouseEventQuirk] undefined initMouseEvent([AtomString] DOMString type,
-        optional boolean canBubble = false, optional boolean cancelable = false,
+    initMouseEvent(optional DOMString type = "undefined", optional boolean canBubble = false, optional boolean cancelable = false,
```

Comment:
WebIDL requires you specify the return type of the function, so:
```suggestion
    undefined initMouseEvent(optional DOMString type = "undefined", optional boolean canBubble = false, optional boolean cancelable = false,
```

Maybe worth filing a bug on the fact that the IDL generation didn't fail the build? This should be a parse error, I think.

Reply:
Thanks @gsnedders 
Trying something

Reply:
> WebIDL requires you specify the return type of the function, so:
> 
> Maybe worth filing a bug on the fact that the IDL generation didn't fail the build? This should be a parse error, I think.

Hi Sam, this is second time where IDL generation didn't failed. I had similar instances in TextTrac / AudioTrack (I can find relevant commit), where we didn't added `readonly` but it compiled and when added, we progressed few WPT. So I think it is worth raising bug.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=268966
@gsnedders feel free to make it clearer.